Remove restangular

furinvader opened the pull request:
Replace restangular with the simpler and more performant `halRequest` service.

https://community.openproject.com/work_packages/23737/activity

furinvader committed:
Add $href property to HalResource

It's intended to replace the current href
property.

furinvader committed:
Implement halRequest service

furinvader committed:
Implement behaviour for null href in halRequest

furinvader committed:
Refactor halResourceTypes config

Rename the function from halResourceTypesStorage
to halResourceTypesConfig

furinvader committed:
Rename halResourceTypesStorage to halResourceFactory

furinvader committed:
Add tests and doc comments to the halResourceFactory

furinvader committed:
Replace apiV3 dependency with halRequest in HalLink

furinvader committed:
Add params option for GET requests to halRequest

furinvader committed:
Allow optional parents to be appended to the path

furinvader committed:
Refactor ProjectService

furinvader committed:
Add v3Path service

furinvader committed:
Remove apiExperimental service

furinvader committed:
Remove console.log from pathBuilder

furinvader committed:
Add headers option to halRequest requests

furinvader committed:
Test apiPaths for appBasePath

furinvader committed:
Add test to v3Path service

furinvader committed:
Remove restangular dependency

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-r-remove-restangular.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://furinvader-r-remove-restangular.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Ignore false/undefined/null values as params in paths

furinvader committed:
Remove apiMetaData service

furinvader committed:
Remove empty params in PathTemplate#callable()

furinvader committed:
Fix error resource handling

furinvader committed:
Correctly reject halRequest promise if there was an error

furinvader committed:
Fix path generation bug

Paths with optional parents saved the templates of
a parent, if the value was set before once.
The path should be generated with the parent only,
if the respective param is provided.

furinvader committed:
Make halRequest return null if the response is empty

furinvader committed:
Always send a request body on halRequest post requests

oliverguenther commented:
Requests are no longer being cached, but instead all are requested. This is especially apparent for schemas and projects, which cause a lot of requests.

![screen shot 2016-08-05 at 08 11 12](https://cloud.githubusercontent.com/assets/459462/17427852/382a34b0-5ae4-11e6-9f5b-b0ad706e5bc0.png)

furinvader committed:
Add caching options to halRequest

oliverguenther commented:
👍
